shared/AppInsightsCore/Tests/Unit/src/StatsBeat.Tests.ts

```diff
@@ -0,0 +1,260 @@
+                        foundExceptionMetric = true;
+                        break;
+                    }
+                }
+                
+                Assert.ok(foundExceptionMetric, "Exception metrics should be tracked");
+            }
+        });
+
+        this.testCase({
+            name: "StatsBeat: does not send metrics for different endpoints",
+            test: () => {
+                // Initialize StatsBeat for a specific endpoint
+                this._statsbeat.initialize(this._core, {
+                    ikey: "Test-iKey",
+                    endpoint: "https://example.endpoint.com",
+                    version: "1.0.0"
+                });
+                
+                // Create mock payload data
+                const payloadData = {
+                    urlString: "https://example.endpoint.com",
+                    data: "testData",
+                    headers: {},
+                    timeout: 0,
+                    disableXhrSync: false,
+                    statsBeatData: {
+                        startTime: Date.now()
+                    }
+                } as IPayloadData;
+                
+                // Set up spies to check internal calls
+                const countSpy = this.sandbox.spy(this._statsbeat, "count");
+                
+                // Count metrics for a different endpoint
+                this._statsbeat.count(200, payloadData, "https://different.endpoint.com");
+                
+                // The count method was called, but it should return early
+                Assert.equal(1, countSpy.callCount, "count method should be called");
+                Assert.equal(0, this._trackSpy.callCount, "track should not be called for different endpoint");
+            }
+        });
+
+        this.testCase({
+            name: "StatsBeat: test dynamic configuration changes",
+            useFakeTimers: true,
+            test: () => {
+                // Setup core with statsbeat enabled
+                this._core.initialize(this._config, [new ChannelPlugin()]);
+                
+                // Verify that statsbeat is created
+                const statsbeat = this._core.getStatsBeat();
+                Assert.ok(statsbeat, "Statsbeat should be created");
+                
+                // Update configuration to disable statsbeat
+if (!this._core.config._sdk) {
+                this._core.config._sdk = {};
+                }
+                this._core.config._sdk.stats = false;
+                this.clock.tick(1); // Allow time for config changes to propagate
+                
+                // Verify that statsbeat is removed
+                const updatedStatsbeat = this._core.getStatsBeat();
+                Assert.ok(!updatedStatsbeat, "Statsbeat should be removed when disabled");
+                
+                // Re-enable statsbeat
+                this._core.config._sdk.stats = true;
```

Comment:
Lets also test setting to a non-boolean (like `null` and `undefined`) to make sure we behave and don't throw.